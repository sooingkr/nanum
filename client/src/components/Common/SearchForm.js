import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { reduxForm, Field } from 'redux-form';
import { Form, Button } from 'react-bootstrap';
import { FoodSearchDuck } from '../../containers/FoodSearch/FoodSearchDuck';

let SearchForm = ({ handleSubmit, onExpand }) => (
  <div className="search-form">
    <Form horizontal onSubmit={handleSubmit}>
      <Field name="foodQuery"
            className="search-form__input"
            component="input"
            type="text"
      />
      <Button bsStyle="default" 
              type="submit"
              className="search-form__submit" 
              onClick={onExpand} >
      </Button>
    </Form>
  </div>
);

const validate = values => {
  const errors = {};
  const { foodQuery } = values;

  if (!foodQuery) {
    errors.foodQuery = 'Please type a food name';
  }

  return errors;
};

SearchForm = reduxForm({
  form: 'SearchForm',
  validate,
})(SearchForm);

// Get initial input value from redux store
const mapStateToProps = (state) => ({
  initialValues: { 
    foodQuery: state[FoodSearchDuck.storeName].foodQuery,
  }
});

SearchForm = connect(mapStateToProps)(SearchForm);

SearchForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
}

export default SearchForm;