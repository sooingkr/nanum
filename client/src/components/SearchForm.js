import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { reduxForm, Field } from 'redux-form';
import { Form, Button } from 'react-bootstrap';

class SearchFormView extends Component {
  handleEnter = (e) => {
    if(e.keyCode === 13 && e.shiftKey === false) {
      this.props.handleSubmit(this.props.values);
    }
  }
  
  render() {
    const { handleSubmit, 
      reset, 
      theme="light", 
      onExpand,
      onClose, 
      isExpanded } = this.props;

    const expandClass = isExpanded ? 'is-expanded' : ''; 

    return (
      <div className={`search-form search-form--${theme} ${expandClass}`}>
        <Form horizontal onSubmit={handleSubmit}>
          <Field name="foodQuery"
                className="search-form__input"
                component="input"
                type="text"
                onKeyDown={this.handleEnter}
          />
          <Button bsStyle="default" 
                  type="button"
                  className="search-form__submit" 
                  onClick={onExpand} >
            <i className="fa fa-search" aria-hidden="true"></i>
          </Button>
          <Button bsStyle="default"
                  type="button"
                  className="search-form__close" 
                  onClick={() => { reset(); onClose();} }>
            <i className="fa fa-times" aria-hidden="true"></i>
          </Button>
          <Button type="submit" style={{visibility: "hidden", padding: 0, border:'none'}}/>
        </Form>
      </div>
    );
  }
}

const validate = values => {
  const errors = {};
  const { foodQuery } = values;

  if (!foodQuery) {
    errors.foodQuery = 'Please type a food name';
  }

  return errors;
};

const SearchForm = reduxForm({
  form: 'search',
  validate,
})(SearchFormView);

SearchForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  theme: PropTypes.oneOf(["light", "dark"]),
}

export default SearchForm;