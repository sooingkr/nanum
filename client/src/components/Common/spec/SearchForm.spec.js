import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import {Form, Field, Button} from 'react-bootstrap';

import SearchForm from '../SearchForm';

describe('SearchForm component', () => {
  const searchWrapper = shallow(<SearchForm />);

  it('should render <SearchForm />', () => {
    expect(toJson(searchWrapper)).toMatchSnapshot();
  });

  it('should render div class="search-form"', () => {
    expect(searchWrapper.find('.search-form').length).toBe(1);
  });

  it('should render a form', () => {
    expect(searchWrapper.find(Form).length).toBe(1);
  });

  it('should render input text', () => {
    expect(searchWrapper.find(Field).length).toBe(1);
  });

  it('should render 2 buttons', () => {
    expect(searchWrapper.find(Button).length).toBeGreaterThanOrEqual(2);
  });
});


