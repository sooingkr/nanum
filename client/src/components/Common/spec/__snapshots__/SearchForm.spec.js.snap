// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component should render <SearchForm /> 1`] = `
<Connect(Form(SearchFormView))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="search"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  validate={[Function]}
/>
`;
